src/vm/database/sqlite.rs

```diff
@@ -297,22 +324,88 @@ impl <'a> ContractDatabase <'a> {
     }
 
     pub fn get_simmed_block_height(&self) -> Result<i128> {
-        let block_height: (i128) =
+        let block_height: (i64) =
             self.query_row(
-                "SELECT simmed_block_height FROM simmed_block_table LIMIT 1",
+                "SELECT block_height FROM simmed_block_table ORDER BY block_height DESC LIMIT 1",
                 NO_PARAMS,
                 |row| row.get(0))
             .expect("Failed to fetch simulated block height");
 
-        Ok(block_height)
+        Ok(block_height as i128)
     }
 
-    pub fn set_simmed_block_height(&mut self, block_height: i128) {
-        self.execute(
-            "UPDATE simmed_block_table SET simmed_block_height = ?",
-            &[block_height]);
+    pub fn get_simmed_block_time(&self, block_height: &i128) -> Result<i128> {
+        let block_time: (i64) = 
+            self.query_row(
+                "SELECT block_time FROM simmed_block_table WHERE block_height = ? LIMIT 1",
+                &[*block_height as i64],
```

Comment:
You should not do this without first checking that the block height fits into `i64`.  If it doesn't, you should return an `Err(...)`.